src/platform/logging/LogV.h

```diff
@@ -30,6 +30,25 @@ namespace Platform {
  */
 void ENFORCE_FORMAT(3, 0) LogV(const char * module, uint8_t category, const char * msg, va_list v);
 
+/**
+ * Open a log file for appending log messages. This function opens
+ * the specified file and ensures that subsequent logs are written to
+ * this file. It should be called at the start of the application
+ * before any logging is done.
+ *
+ * @param[in] filePath  The path to the log file. If the file does not
+ *                      exist, it will be created. If it exists, log
+ *                      messages will be appended to it.
+ */
+void OpenLogFile(const char * filePath);
```

Comment:
We should have a list of log sinks (one of which is stdout by default) rather than assume a single place where logs go. Otherwise it is impossible to have stdout + output to file logs.

Reply:
We do currently allow setting one log sink (the logging redirector setup), which can then be used to multiplex as desired without changing any platform implementations....

Reply:
https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/logging/TextOnlyLogging.h#L359 contains a `SetLogRedirectCallback` where you can move your logging to go anywhere. This is what python uses to redirect to python logging instead of just console.

Could we use this for the CLI bits instead? One could even imagine some term-linke thing where you can both do cli-shell-interact and view the log in a separate pane...

Reply:
Yes, this method works.